lightning/src/ln/offers_tests.rs

```diff
@@ -512,6 +512,12 @@ fn creates_and_pays_for_offer_using_two_hop_blinded_path() {
 	create_announced_chan_between_nodes_with_value(&nodes, 2, 4, 10_000_000, 1_000_000_000);
 	create_announced_chan_between_nodes_with_value(&nodes, 2, 5, 10_000_000, 1_000_000_000);
 
+	// Create an announced channel between Bob and node[6] to ensure that in the event of a tie
+	// between Bob and Charlie for becoming the introduction node candidate, Bob is deterministically
+	// preferred over Charlie.
+	create_announced_chan_between_nodes_with_value(&nodes, 1, 6, 10_000_000, 1_000_000_000);
```

Comment:
Is there a reason why this is needed here but not in other tests? Instead of doing this, could we instead call `disconnect_peers` as/when needed to avoid the non-determinism?

Reply:
I see, thanks for explaining. So it isn't failing in other cases because they are either only one hop, and thus have a smaller network or is the refund case, which has less interaction.

Anyhow, we can avoid adding another node and channel to the network by changing the assert to pass for either possible introduction node.

```rust
assert!(
	matches!(
		reply_path.introduction_node(),
		&IntroductionNode::NodeId(node_id) if node_id == bob_id || node_id == charlie_id,
	)
);
```

Reply:
Definitely looks like a cleaner solution!
Updated the check, and removed the extra channel in [pr3163.07](https://github.com/shaavan/rust-lightning/commits/pr3163.07)
Thanks a lot for the suggestion!